base/src/main/java/org/aya/resolve/context/ModuleSymbol.java

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2020-2023 Tesla (Yinsen) Zhang.
+// Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+package org.aya.resolve.context;
+
+import kala.collection.Map;
+import kala.collection.MapView;
+import kala.collection.SetView;
+import kala.control.Option;
+import kala.control.Result;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+
+public interface ModuleSymbol<T extends ContextUnit> {
+
+  /**
+   * Getting the candidates of an unqualified name
+   *
+   * @param unqualifiedName the unqualified name
+   * @return the candidates, probably empty
+   */
+  @NotNull Map<ModulePath, T> getCandidates(@NotNull String unqualifiedName);
```

Comment:
`resolveUnqualified` looks like a better name: it implies there will be candidates and the input name is unqualified.